core/src/main/java/org/apache/calcite/rel/rules/SubQueryRemoveRule.java

```diff
@@ -926,6 +923,11 @@ private static void matchJoin(SubQueryRemoveRule rule, RelOptRuleCall call) {
     if (inputIntersectsLeftSide) {
       builder.push(join.getLeft());
 
+      final RelOptUtil.Logic logic =
```

Comment:
this code could use a little comment.
(I am not familiar with this part of the code...)